src/beam_psrfits.c

```diff
@@ -145,12 +156,21 @@ void populate_spliced_psrfits_header(
 
     int i; // idx into dat_freqs
     int iF, iC; // mwalib (i)dxs for (F)ine and (C)oarse channels
-    uint32_t start_hz = vm->obs_metadata->metafits_coarse_chans[0].chan_start_hz;
+    uint32_t start_hz = vm->obs_metadata->metafits_coarse_chans[first_coarse_chan_idx].chan_start_hz;
+
     for (i = 0 ; i < pf->hdr.nchan; i++)
     {
-        iC = i / vm->nfine_chan + first_coarse_chan_idx;
+        /* BWM comment, 20 Oct 2023:
+         * Previously, the addition of the `first_coarse_chan_index` to `iC` caused an offset
+         * proportional to that coarse channel number (multiplied by the fine channel width).
+         * This is fine when it is a contiguous band because that index is always zero! But in
+         * picket-fence mode, because the starting frequency is already defined via "first
+         * coarse channel" number, we are only ever concerned with the fine channel additions
+         * in this loop.
+         */
+        iC = i / vm->nfine_chan;
```

Comment:
So this is the main thing to change to get it work right? Everything elsewhere seems like all stylistic changes

Reply:
Yes, this was the primary culprit I believe.